internal/status/gatewayclassconditions_test.go

```diff
@@ -180,23 +170,27 @@ func TestMergeConditions(t *testing.T) {
 				newCondition("available", "false", "Reason", "Message", start),
 			},
 			updates: []metav1.Condition{
-				newCondition("available", "false", "Reason", "New Message", middle),
+				newCondition("available", "false", "Reason", "New Message", later),
 			},
 			expected: []metav1.Condition{
 				newCondition("available", "false", "Reason", "New Message", start),
 			},
 		},
+		{
+			name:    "new status",
+			current: []metav1.Condition{},
+			updates: []metav1.Condition{
+				newCondition("available", "false", "Reason", "New Message", later),
+			},
+			expected: []metav1.Condition{
+				newCondition("available", "false", "Reason", "New Message", later),
+			},
+		},
 	}
 
-	// Simulate the passage of time between original condition creation
-	// and update processing
-	fakeClock.SetTime(later)
-
 	for _, tc := range testCases {
 		got := mergeConditions(tc.current, tc.updates...)
-		if conditionChanged(tc.expected[0], got[0]) {
-			assert.Equal(t, tc.expected, got, tc.name)
-		}
+		assert.Equal(t, tc.expected, got, tc.name)
```

Comment:
rather than use `conditionChanged` which doesnt check the transition time, just use `assert.Equal` to check equality

also means as per how the tests were originally set up, we needed to change the implementation to only update the transition time if the status changed